src/Common/src/Interop/OSX/System.Native/Interop.ProcRUsage.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        // Constants from sys/resource.h
+        private const int RUSAGE_SELF = 0;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct RUsageInfo
```

Comment:
These structs do have a fixed layout and are OS X only (as far as I know). This pack started off abstracting away how generic Process Information was retrieved and passing back an intermediate struct, but after we decided to go in a different direction I figured I could just check in the shim'd OS X portions.